x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/fields/monitor_spaces.tsx

```diff
@@ -0,0 +1,149 @@
+  allSpacesId?: string
+): string[] {
+  if (allSpacesId && selectedIds.includes(allSpacesId)) {
+    // Only return allSpacesId if selected, ignore all others including currentSpaceId
+    return [allSpacesId];
+  }
+  // Remove allSpacesId if present (should only be present alone)
+  const filtered = allSpacesId ? selectedIds.filter((id) => id !== allSpacesId) : selectedIds;
+  if (filtered.length === 0 && currentSpaceId) {
+    return [currentSpaceId];
+  }
+  if (currentSpaceId && !filtered.includes(currentSpaceId)) {
+    return [...filtered, currentSpaceId];
+  }
+  return filtered;
+}
+
+export const MonitorSpaces = ({ value, onChange, ...rest }: MonitorSpacesProps) => {
+  const { space: currentSpace } = useKibanaSpace();
+  const NAMESPACES_NAME = 'spaces';
+  const { services } = useKibana<ClientPluginsStart>();
+  const [spacesList, setSpacesList] = React.useState<Array<{ id: string; label: string }>>([]);
+  const data = services.spaces?.ui.useSpaces();
+
+  const {
+    control,
+    formState: { isSubmitted },
+    trigger,
+  } = useFormContext();
+  const { isTouched, error } = control.getFieldState(NAMESPACES_NAME);
+
+  const showFieldInvalid = (isSubmitted || isTouched) && !!error;
+
+  useEffect(() => {
+    if (data?.spacesDataPromise) {
+      data.spacesDataPromise.then((spacesData) => {
+        setSpacesList([
+          allSpacesOption,
+          ...[...spacesData.spacesMap].map(([spaceId, dataS]) => ({
+            id: spaceId,
+            label: dataS.name,
+          })),
+        ]);
+      });
+    }
+  }, [data]);
+
+  useEffect(() => {
+    // set space as current value if no value is provided
+    if ((!value || value.length === 0) && currentSpace) {
+      onChange([currentSpace.id]);
+    }
+  }, [onChange, currentSpace, value]);
+
+  // Ensure selected options always include the current space
+  const selectedIds = React.useMemo(() => {
+    if (!currentSpace) {
+      return value ?? [];
+    }
+    if (!value || value.length === 0) {
+      return [currentSpace.id];
+    }
+    return value.includes(currentSpace.id) || value.includes(ALL_SPACES_ID)
+      ? value
+      : [...value, currentSpace.id];
+  }, [value, currentSpace]);
+
+  // Compute if "All spaces" is selected
+  const isAllSpacesSelected = selectedIds.includes(ALL_SPACES_ID);
+
+  return (
+    <EuiComboBox<string>
+      fullWidth
+      aria-label={SPACES_LABEL}
+      placeholder={SPACES_LABEL}
+      isInvalid={showFieldInvalid}
+      onBlur={async () => {
+        await trigger();
+      }}
+      options={spacesList.map((option) =>
+        isAllSpacesSelected && option.id !== ALL_SPACES_ID ? { ...option, disabled: true } : option
+      )}
+      selectedOptions={selectedIds.map((id) => {
+        const sp = spacesList.find((spaceObj) => spaceObj.id === id);
+        if (!sp) {
+          return {
+            id,
+            label: id,
+          };
+        }
+        return { id: sp.id, label: sp.label };
```

Comment:
Isn't this the same?

```suggestion
      selectedOptions={spacesList.filter(({ id }) => selectedIds.includes(id))}
```